src/model/resources/resources.jl

```diff
@@ -1,180 +1,895 @@
+
+Returns the indices of all hydro resources in the vector `rs`.
+"""
+hydro(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,Hydro), rs)
+
+# THERM interface
+"""
+    thermal(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all thermal resources in the vector `rs`.
+"""
+thermal(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,Thermal), rs)
+up_time(r::Thermal) = get(r, :up_time, default_zero)
+down_time(r::Thermal) = get(r, :down_time, default_zero)
+pwfu_fuel_usage_zero_load_mmbtu_per_h(r::Thermal) = get(r, :pwfu_fuel_usage_zero_load_mmbtu_per_h, default_zero)
+
+# VRE interface
+"""
+    vre(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all Vre resources in the vector `rs`.
+"""
+vre(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,Vre), rs)
+
+# ELECTROLYZER interface
+"""
+    electrolyzer(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all electrolyzer resources in the vector `rs`.
+"""
+electrolyzer(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,Electrolyzer), rs)
+electrolyzer_min_kt(r::Electrolyzer) = r.electrolyzer_min_kt
+hydrogen_mwh_per_tonne(r::Electrolyzer) = r.hydrogen_mwh_per_tonne
+hydrogen_price_per_tonne(r::Electrolyzer) = r.hydrogen_price_per_tonne
+
+# FLEX_DEMAND interface
+"""
+    flex_demand(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all flexible demand resources in the vector `rs`.
+"""
+flex_demand(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,FlexDemand), rs)
+flexible_demand_energy_eff(r::FlexDemand) = r.flexible_demand_energy_eff
+max_flexible_demand_delay(r::FlexDemand) = r.max_flexible_demand_delay
+max_flexible_demand_advance(r::FlexDemand) = r.max_flexible_demand_advance
+var_om_cost_per_mwh_in(r::FlexDemand) = get(r, :var_om_cost_per_mwh_in, default_zero)
+
+# MUST_RUN interface
+"""
+    must_run(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all must-run resources in the vector `rs`.
+"""
+must_run(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,MustRun), rs)
+
+# VRE_STOR interface
+"""
+    vre_stor(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all VRE_STOR resources in the vector `rs`.
+"""
+vre_stor(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,VreStorage), rs)
+technology(r::VreStorage) = r.technology
+self_discharge(r::VreStorage) = r.self_disch
+
+"""
+    solar(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all co-located solar resources in the vector `rs`.
+"""
+solar(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,VreStorage) && r.solar != 0, rs)
+
+"""
+    wind(rs::Vector{T}) where T <: AbstractResource
+
+Returns the indices of all co-located wind resources in the vector `rs`.
+"""
+wind(rs::Vector{T}) where T <: AbstractResource = findall(r -> isa(r,VreStorage) && r.wind != 0, rs)
+
+"""
```

Comment:
I'm not sure the best way to fix this yet, but it might be confusing to readers if `solar` and `wind` only pick up `VreStorage` resources and not `Vre`s.